src/mpc-hc/MainFrm.cpp

```diff
@@ -8196,6 +8196,12 @@ void CMainFrame::OnUpdatePlayPauseStop(CCmdUI* pCmdUI)
                     fEnable = false;
                 }
             }
+
+            if (m_media_trans_control.controls) {

```

Comment:
This GUI function gets called very often. So it is better to move the status change to the places where the state actually changes: MediaControlRun()/Pause()/Stop().

Also refactor into a function: MediaTransControlUpdateState(OAFilterState state);